src/sqlfluff/api/simple.py

```diff
@@ -123,8 +123,17 @@ def fix(
     linter = Linter(config=cfg)
 
     result = linter.lint_string_wrapped(sql, fix=True)
-    fixed_string = result.paths[0].files[0].fix_string()[0]
-    return fixed_string
+    fix_even_unparsable = cfg.get("fix_even_unparsable")
+    should_fix = True
+    if not fix_even_unparsable:
+        # If fix_even_unparsable wasn't set, check for parse errors. Okay to fix
+        # if there are no parse errors.
+        _, num_filtered_parse_errors = result.check_parse_errors()
+        if num_filtered_parse_errors == 0:
+            should_fix = True
```

Comment:
Doesn’t this need to be `!=` and also setting it to false?

btw do like this better. Though personally think I would have gone for parse error check first and then see if needed. Even if it’s a slight waste of processing for those using the force flag.

Reply:
Ah yes. Cuz it's already set to true. The risk of refactoring working code. 😉